pandas/core/internals/managers.py

```diff
@@ -872,7 +872,9 @@ def as_array(
 
         return arr.transpose() if transpose else arr
 
-    def _interleave(self, dtype=None, na_value=lib.no_default) -> np.ndarray:
+    def _interleave(
+        self, dtype: Optional[Dtype] = None, na_value=lib.no_default
```

Comment:
NpDtype

Reply:
On L889 we do `elif is_extension_array_dtype(dtype):`. Based on this I thought `Dtype` was more appropriate here, since `NpDtype` does not include ExtensionArray Dtypes